components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt

```diff
@@ -102,7 +102,7 @@ class DefaultTopSitesStorage(
             try {
                 providerTopSites = topSitesProvider
                     .getTopSites(allowCache = true)
-                    .take(numSitesRequired)
+                    .take(providerConfig.maxThreshold - pinnedSites.size)
```

Comment:
just to make sure I'm on the same page, the math here is to ensure enough top sites are grabbed so long as there's room for them?

Reply:
So, the maxThreshold is 8 (meaning we don't want to take any provided top sites if there are more than 8 pinned sites). The prior if statements guarantee that we have less pinned sites than the maxThreshold. On Fenix, we want to ensure provided top sites are only shown in the first page (within the first 8 top sites). So, if there are 8 pinned sites, we show no provided top sites. If it's 7 pinned sites, we only show 1 provided top sites. So, the math ends up being maxThreshold - pinnedSites.size

Reply:
`take(n)` with a negative number would error, but it's not possible because of the check in the if statement to ensure pinnedSites.size is less than the maxThreshold